### PR TITLE
Us770573/new psi trigger

### PR DESCRIPTION
**Pull request by pavelzlatnik**
Jenkins Trigger modification to support creation of PSI for ZOWE together with SMPE creation 

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [x] Feature 
- [ ] Other... Please describe:

#### Changes proposed in this PR
Modification of Jenkins trigger to support also PSI creation from SMPE environment
(Now only with dummy PSI script which just echo message. Real script that creates PSI would be added after check that pipeline is working as expected).
- 

#### Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: smpe/bld/smpe.sh [hunk 1]
```diff
@@ -169,7 +170,8 @@ do case "$opt" in
   P)   export reqPTF="-P";;
   p)   export VERSION="-p $OPTARG";;
   r)   export ROOT="$OPTARG";;
-  s)   export stopAt="$OPTARG";;
+  S)   export psi="$OPTARG";;
```
Comment (OnnoVdT) [inline smpe/bld/smpe.sh:173]: use the following when not using a second parm for -S
`S)  export psi="-S";;`

File: smpe/bld/smpe.sh [hunk 2]
```diff
@@ -272,6 +274,12 @@ _stopAt smpe-service.sh $debug -c $YAML $opts
 _cmd $here/smpe-service.sh $debug -c $YAML $opts
 # result (final): $ship                      # zip with sysmod & readme
 
+# create psi
```
Comment (OnnoVdT) [inline smpe/bld/smpe.sh:277]: the whole code block should be wrapped in the if
`# create psi`
`if test -n "$psi"`
`then`
`  opts=""`
`  _stopAt prepare-PSI.sh $debug -c $YAML $opts`
`  _cmd $here/psi/bld/prepare-PSI.sh $debug -c $YAML $opts`
  
`  # TODO create PSI`
`fi    # $psi`

File: smpe/bld/psi/bld/prepare-PSI.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
```
Comment (OnnoVdT) [inline smpe/bld/psi/bld/prepare-PSI.sh:1]: it will be easier to reuse existing smpe-*.sh code if the psi scripts are in the same dir as get-config.sh

**Review by OnnoVdT — CHANGES_REQUESTED** [review]

**Comment by OnnoVdT** [comment]
To answer the question asked in #2282 before it was closed, 
smpe-install-smpe.sh is a new script to install the SMPE verson (FMID, or FMID + PTF if there is a PTF). To be consistent, the existing smpe-install.sh could be renamed to smpe-install-pax.sh to clearly show what is being installed now that there will be two.

File: smpe/bld/smpe.sh [hunk 4]
```diff
@@ -29,6 +29,7 @@
 #% -r rootDir    use the specified root directory
 #%               ignored when -c is specified
 #% -s stopAt.sh  stop before this sub-script is invoked          #debug
+#% -S psi.sh     execute script to create zowe PSI
```
[the remark this answers is not in the document]
Comment (pavelzlatnik) [reply smpe/bld/smpe.sh]: yes, this has more sense

File: smpe/bld/smpe.sh [hunk 5]
```diff
@@ -155,7 +156,7 @@ unset alter BUILD BRANCH YAML SuCcEsS count HLQ input reqPTF VERSION \
 errorRC=8  # default RC 8 on error
 
 # get startup arguments
-while getopts a:B:b:c:E:f:h:i:p:r:s:V:v:1:2:?dP opt
+while getopts a:B:b:c:E:f:h:i:p:r:S:s:V:v:1:2:?dP opt
```
[the remark this answers is not in the document]
Comment (pavelzlatnik) [reply smpe/bld/smpe.sh]: corrected

File: smpe/bld/smpe.sh (hunk 1 above)
Comment (pavelzlatnik) [reply smpe/bld/smpe.sh]: updated

File: smpe/bld/smpe.sh (hunk 2 above)
Comment (pavelzlatnik) [reply smpe/bld/smpe.sh:277]: Updated based on feedback

File: smpe/bld/psi/bld/prepare-PSI.sh (hunk 3 above)
Comment (pavelzlatnik) [reply smpe/bld/psi/bld/prepare-PSI.sh]: ok, I have moved psi script to the same directory

File: smpe/bld/smpe.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (OnnoVdT) [reply smpe/bld/smpe.sh]: step 2 is incomplete, you removed S:, in the middle, but did not add S at the end

**Comment by OnnoVdT** [comment]
I realize now I forgot to mention with last review that it would make sense to update .pax/post.packaging.sh in this PR as well. Here we have to add the -S parm to the invocation of smpe.sh if the PSIANDSMPE option was set in Jenkins.
`# start SMPE build`
...
`#% -S              create PSI package (Portable Software Instance)`
...
`echo "BUILD_PSI=$BUILD_PSI"`
`test -n "$BUILD_PSI" && external="$external -S"`

**Review by OnnoVdT — APPROVED** [review]

**Comment by jackjia-ibm** [comment]
About to code freeze, will merge it. Please submit new PR to fix any issues. Thank you.

**Pull request merged**